### DISCUSSION TITLE
Auto save on a vanilla js template shows wrong browser preview data

### DISCUSSION THREAD

**Discussion by skillsboy** [codesandbox/codesandbox-client#5611] [General] [answered]
When I have the auto save option on and change a vanilla js template that adds html to the dom via js the preview browser changes to the new data manipulated by the js file but after that (1s) it shows the browser preview with only the pure html from the html file without any dom manipulation. This also happens if I have the auto save option off and save twice (ctrl + s)

Would like to know if I'm doing something wrong?

index.html
```
<!DOCTYPE html>
<html>
  <head>
    <title>Parcel Sandbox</title>
    <meta charset="UTF-8" />
  </head>

  <body>
    <div id="app">No manipulation</div>

    <script src="index.js"></script>
  </body>
</html>
```

index.js
`document.getElementById("app").innerHTML = "<h1>Hello Vanilla!</h1>"`

**Comment by smhutch** [comment, marked as answer]
@skillsboy I could replicate this issue. The most curious part about this is that saving a file twice has a different result on the first and second save.

I'm unsure of exactly what is causing this, but am also interested in knowing why you are looking to add text ("No manipulation") inside the `div` tag? I would recommend either adding the HTML you require to `index.js`, or removing the `id="app"` from your HTML (which is used to replace the contents).

**Reply by skillsboy** [reply]
Thank you for the reply @smhutch .

The text ("No manipulation") was just for a test reason, was not intending to use it like this :).

Thanks for the advice, I appreciate it!
